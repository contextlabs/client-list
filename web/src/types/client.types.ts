
export interface Client {
    readonly name: string;
    readonly title: string;
    readonly avatar: string;
    readonly quote: string;
    readonly nationality: string;
}

