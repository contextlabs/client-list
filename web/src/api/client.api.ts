import { Client } from '../types/client.types';
import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
import { DEFAULT_AXIOS_CONFIG } from '../constants';

export const PATHS = Object.freeze({
  CLIENT_DATA: () => '/clients'
});

const BASE_URL = 'http://localhost:5000';

const CONFIG: AxiosRequestConfig = {
  ...DEFAULT_AXIOS_CONFIG,
  baseURL: BASE_URL
};

export const getClients = (): Promise<AxiosResponse<Client[]>> => {
  return axios.get<Client[]>(PATHS.CLIENT_DATA(), CONFIG);
};

export const getClientByName = (
  name: string
): Promise<AxiosResponse<Client[]>> => {
  const params = { name };
  return axios.get<Client[]>(PATHS.CLIENT_DATA(), { ...CONFIG, params });
};

export const getClientByFilter = (
  q: string
): Promise<AxiosResponse<Client[]>> => {
  const params = { q };
  return axios.get<Client[]>(PATHS.CLIENT_DATA(), { ...CONFIG, params });
};
