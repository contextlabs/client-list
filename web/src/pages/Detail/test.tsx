import { render } from '@testing-library/react';
import { Client } from '../../types/client.types';
import { Detail } from '.';
import * as api from '../../api/client.api';
import { AxiosResponse } from 'axios';
import { act } from 'react-dom/test-utils';
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => mockedUsedNavigate
}));

const MOCK_DATA: AxiosResponse<Client[]> = {
  data: [
    {
      name: 'CardDetail Name',
      title: 'CardDetail Title',
      avatar: 'https://gravatar.com/4405735f6f3129e0286d9d43e7b460d0',
      nationality: 'Global',
      quote: 'Hello'
    }
  ],
  status: 0,
  statusText: '',
  headers: {},
  config: {}
};

describe('<Detail />', () => {
  it('should render the Detail as default', async () => {
    jest.spyOn(api, 'getClientByName').mockResolvedValue(MOCK_DATA);
    const { container } = render(<Detail />);
    await act(async () => {
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
