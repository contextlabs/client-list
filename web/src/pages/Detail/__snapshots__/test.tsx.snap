// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail /> should render the Detail as default 1`] = `
<div
  class="p-6 w-full bg-white rounded-lg border-l-4 border-teal-500 border-solid shadow-md"
>
  <div>
    Loading ...
  </div>
</div>
`;
